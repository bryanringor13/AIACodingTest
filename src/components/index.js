// General Component
export {default as Container} from './general/Container';
export {default as Center} from './general/Box/Center';
export {default as Box} from './general/Box';
export {default as TextInput} from './general/TextInput';
export {default as Button} from './general/Button';
export {default as Image} from './general/Image';
export {default as ImageAssIcon} from './general/Image/Assesment';
export {default as Text} from './general/Text';
export {default as Link} from './general/Link';

// Card Component
export {default as Card} from './general/Card';
export {default as Content} from './general/Card/Content';

// Login Component
export {default as BottomModal} from './login/BottomModal';
export {default as CustomBackdrop} from './login/BottomModal/CustomBackdrop';
