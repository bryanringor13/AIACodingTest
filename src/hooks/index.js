export {UserProvider} from './user/index';
