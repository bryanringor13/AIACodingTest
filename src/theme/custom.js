export const currentTheme = {
  fontSize: {
    sm: '12px',
    md: '16px',
    lg: '18px',
    xl: '24px',
  },
  color: {
    danger: '#df4759',
    success: '#42ba96',
    link: '#0000EE',
  },
};
