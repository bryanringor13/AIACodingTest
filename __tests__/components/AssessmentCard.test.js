import React from 'react';
import renderer from 'react-test-renderer';
import {ThemeProvider} from 'styled-components/native';
import {Content} from '../../src/components';
import {currentTheme} from '../../src/theme/custom';
import {SCREEN_WIDTH} from '../../src/utils/constant';

global.fetch = jest.fn(() => new Promise(resolve => resolve()));

describe('Assessment Card Component', () => {
  const width = SCREEN_WIDTH * 0.7;
  const assessmentWPtsData = {
    id: '0001',
    title: 'Advanced Health Screening',
    pts: 99999,
  };

  const assessmentWOPtsData = {
    id: '0001',
    title: 'Advanced Health Screening',
  };

  it('With 99999 points property', () => {
    const withPts = renderer
      .create(
        <ThemeProvider theme={currentTheme}>
          <Content item={assessmentWPtsData} {...{width}} />
        </ThemeProvider>,
      )
      .toJSON();
    expect(withPts).toMatchSnapshot();
  });

  it('Without points property', () => {
    const withoutPts = renderer
      .create(
        <ThemeProvider theme={currentTheme}>
          <Content item={assessmentWOPtsData} {...{width}} />
        </ThemeProvider>,
      )
      .toJSON();
    expect(withoutPts).toMatchSnapshot();
  });
});
