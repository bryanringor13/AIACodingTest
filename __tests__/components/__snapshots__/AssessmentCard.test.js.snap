// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Assessment Card Component With 99999 points property 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#FFFFFF",
        "borderRadius": 5,
        "marginLeft": 5,
        "marginRight": 10,
        "marginTop": 10,
        "paddingBottom": 10,
        "paddingLeft": 10,
        "paddingRight": 10,
        "paddingTop": 10,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "2625": "column",
          "height": 200,
          "justifyContent": "space-between",
          "marginLeft": 5,
        },
      ]
    }
    width={262.5}
  >
    <View>
      <Text
        style={
          Array [
            Object {
              "fontSize": 16,
              "fontWeight": "bold",
              "marginBottom": 5,
              "marginTop": 10,
            },
          ]
        }
      >
        Advanced Health Screening
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginTop": 10,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "width": "60%",
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "gray",
              },
            ]
          }
        >
          Earn up to 
          <Text
            style={
              Array [
                Object {
                  "color": "rgb(56, 110, 157)",
                  "fontSize": 16,
                  "fontWeight": "bold",
                },
              ]
            }
          >
            99999
          </Text>
           pts
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "flex-end",
              "justifyContent": "flex-end",
              "width": "40%",
            },
          ]
        }
      >
        <Text
          allowFontScaling={false}
          selectable={false}
          style={
            Array [
              Object {
                "color": "rgb(123,32,76)",
                "fontSize": 18,
              },
              undefined,
              Object {
                "fontFamily": "anticon",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          î˜›
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`Assessment Card Component Without points property 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#FFFFFF",
        "borderRadius": 5,
        "marginLeft": 5,
        "marginRight": 10,
        "marginTop": 10,
        "paddingBottom": 10,
        "paddingLeft": 10,
        "paddingRight": 10,
        "paddingTop": 10,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "2625": "column",
          "height": 200,
          "justifyContent": "space-between",
          "marginLeft": 5,
        },
      ]
    }
    width={262.5}
  >
    <View>
      <Text
        style={
          Array [
            Object {
              "fontSize": 16,
              "fontWeight": "bold",
              "marginBottom": 5,
              "marginTop": 10,
            },
          ]
        }
      >
        Advanced Health Screening
      </Text>
    </View>
  </View>
</View>
`;
